controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -681,3 +938,49 @@ func diffAddresses(oldAddresses, newAddresses AddressSet) (add, remove AddressSe
 	}
 	return
 }
+
+func isTargetPortInSlice(targetPort namedPort, slicePorts []dv1beta1.EndpointPort) bool {
+	for _, p := range slicePorts {
+		switch targetPort.Type {
+		case intstr.Int:
+			if *p.Port == targetPort.IntVal {
+				return true
+			}
+		case intstr.String:
+			if *p.Name == targetPort.StrVal {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+func isValidSlice(es *dv1beta1.EndpointSlice) bool {
+	if es.AddressType != dv1beta1.AddressTypeIPv4 {
```

Comment:
I know that more work is required to properly support IPv6 but do we need to add code to skip non-IPv4 addresses?  This just seems like one more thing we'll need to remove when we properly support IPv6.

Reply:
The probability of an EndpointSlice with an IPv6 address type (or FQDN for that matter) should be quite low. I don't think any would be processed at the moment since IPv6 networking in Kubernetes is not supported out-of-the-box and itself sits behind feature gates, so I can remove this.

If anybody feels like this should be added back in, let me know and I'll handle it.